py3dtiles/b3dm.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+import struct
+import numpy as np
+
+from .tile import Tile, TileHeader, TileBody, TileType
+from .gltf import GlTF
+
+class B3dm(Tile):
+
+    @staticmethod
+    def from_glTF(gltf):
+        """
+        gltf : GlTF
+            glTF object representing a set of objects
+
+        Returns
+        -------
+        tile : Tile
+        """
+
+        tb = B3dmBody()
+        tb.glTF = gltf
+
+        th = B3dmHeader()
+
+        t = Tile()
+        t.body = tb
+        t.header = th
+
+        return t
+
+
+class B3dmHeader(TileHeader):
+    BYTELENGTH = 24
+
+    def __init__(self):
+        self.type = TileType.BATCHED3DMODEL
+        self.magic_value = "b3dm"
+        self.version = 1
+        self.tile_byte_length = 0
+        self.bt_json_byte_length = 0
+        self.bt_bin_byte_length = 0
+        self.bt_length = 0  # number of models in the batch
+
+    def to_array(self):
+        header_arr = np.fromstring(self.magic_value, np.uint8)
+
+        header_arr2 = np.array([self.version,
+                                self.tile_byte_length,
+                                self.bt_json_byte_length,
+                                self.bt_bin_byte_length,
+                                self.bt_length], dtype=np.uint32)
+
+        return np.concatenate((header_arr, header_arr2.view(np.uint8)))
+
+    def sync(self, body):
+        """
+        Allow to synchronize headers with contents.
+        """
+
+        # extract array
+        glTF_arr = body.glTF.to_array()
+        #bth_arr = body.batch_table.header.to_array()
+        #btb_arr = body.batch_table.body.to_array()
+
+        # sync the tile header with feature table contents
+        self.magic_value = "b3dm"
```

Comment:
`magic_value` should not be changed during the lifecycle of the instance (it could be a private parameter to avoid some unfortunate modification). So it shouldn't be necessary to sync this parameter. What do you think?